x/distribution/keeper/allocation.go

```diff
@@ -0,0 +1,63 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+)
+
+// Allocate fees handles distribution of the collected fees
+func (k Keeper) AllocateFees(ctx sdk.Context) {
+	ctx.Logger().With("module", "x/distribution").Error(fmt.Sprintf("allocation height: %v", ctx.BlockHeight()))
+
+	// if there is no power in the system nothing should be allocated
+	bondedTokens := k.stakeKeeper.TotalPower(ctx).TruncateInt()
+	if bondedTokens.IsZero() {
+		return
+	}
+
+	// get the proposer of this block
+	proposerConsAddr := k.GetProposerConsAddr(ctx)
+	proposerValidator := k.stakeKeeper.ValidatorByConsAddr(ctx, proposerConsAddr)
+	proposerDist := k.GetValidatorDistInfo(ctx, proposerValidator.GetOperator())
+
+	// get the fees which have been getting collected through all the
+	// transactions in the block
+	feesCollected := k.feeCollectionKeeper.GetCollectedFees(ctx)
+	feesCollectedDec := types.NewDecCoins(feesCollected)
+
+	// allocated rewards to proposer
+	sumPowerPrecommitValidators := k.GetSumPrecommitPower(ctx)
+	percentVoting := sdk.NewDec(sumPowerPrecommitValidators).QuoInt(bondedTokens)
+
+	// rare edge case for rounding tendermint power vs bonded decimal power
```

Comment:
See the above comment, are you sure rounding is the cause here? Tendermint power should always round *down*.

Reply:
addressed by above